backend/auth/service/login.go

```diff
@@ -0,0 +1,33 @@
+package service
+
+import (
+	"fmt"
+
+	"github.com/jinzhu/gorm"
+	"github.com/sirupsen/logrus"
+	"github.com/stamhoofd/stamhoofd/backend/auth/models"
+	"golang.org/x/crypto/bcrypt"
+)
+
+func Login(db *gorm.DB, email, password string) (*models.User, error) {
+	user := &models.User{}
+	err := db.Where(&models.User{
+		Email: email,
+	}).First(user).Error
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve user %s: %w", email, err)
+	}
+
+	err = bcrypt.CompareHashAndPassword([]byte(user.Password), []byte(password))
+	if err != nil {
+		return nil, fmt.Errorf("login failed: %w", err)
+	}
+
+	err = db.Model(user).Update("registration_token", nil).Error
+	if err != nil {
+		return nil, fmt.Errorf("could not update registration token for user %s: %w", email, err)
+	}
```

Comment:
Already done in `RegisterConfirm`?

Reply:
Maybe instead check if it is empty (if we don't want to allow sign in before confirmation).

Reply:
Yes, I thought the same thing. But I think I somehow want to "private" login function here that only accepts a user and is used by the actual login and RegisterConfirm so both actually log the user in. But this honestly depends on how we are going to do the login and whether we will allow users to login without activating their accounts first.